src/sim/syscall_emul.hh

```diff
@@ -677,6 +678,68 @@ copyOutStatfsBuf(TgtStatPtr tgt, HostStatPtr host)
 #endif
 }
 
+template <typename OS, typename TgtStatPtr, typename HostStatPtr>
+void
+copyOutStatxBuf(TgtStatPtr tgt, HostStatPtr host, bool fakeTTY = false)
+{
+#if !defined(__linux__)
+    static_assert(false, "statx is a Linux-specific system call.");
+#endif
+
+    const auto bo = [] (auto& value) {
+        constexpr ByteOrder bo = OS::byteOrder;
+        return value = htog(value, bo);
+    };
+
+    if (fakeTTY) {
+        tgt->stx_dev_major = MAJOR(0xA);
+        tgt->stx_dev_minor = MINOR(0xA);
+    } else {
+        tgt->stx_dev_major = host->stx_dev_major;
+        tgt->stx_dev_minor = host->stx_dev_minor;
+    }
+    bo(tgt->stx_dev_major);
+    bo(tgt->stx_dev_minor);
+    bo(tgt->stx_ino = host->stx_ino);
+    tgt->stx_mode = host->stx_mode;
+    if (fakeTTY) {
+      // Claim to be character device.
+      tgt->stx_mode &= ~S_IFMT;
+      tgt->stx_mode |= S_IFCHR;
+    }
+    bo(tgt->stx_mode);
+    bo(tgt->stx_nlink = host->stx_nlink);
+    bo(tgt->stx_uid = host->stx_uid);
+    bo(tgt->stx_gid = host->stx_gid);
```

Comment:
I honestly doubt

    bo(tgt->stx_nlink = host->stx_nlink)

is more readable and intuitive than

    tgt->stx_nlink = htog(host->stx_nlink, bo);

Especially as it relies on some obscure C assignment rule on what an assignment expression evaluates to. I'd vote to keep this simple

Reply:
I changed it in the latest commit to use the method used in other stat* functions.